### PR TITLE
[ODS-4488] Fix broken composite features reported by the postman tests

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs [hunk 1]
```diff
@@ -788,7 +788,7 @@ private void ProcessQueryStringParameters(HqlBuilderContext builderContext, Comp
                     {
                         var id = _descriptorsCache.GetId(
                             targetProperty.LookupTypeName,
-                            (string) queryStringParameter.Value);
+                            Convert.ToString(queryStringParameter.Value));
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:791]: Why the change from the cast?

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs [hunk 2]
```diff
@@ -797,7 +797,7 @@ private void ProcessQueryStringParameters(HqlBuilderContext builderContext, Comp
                     // Handle UniqueId conversions
                     else if (UniqueIdSpecification.TryGetUniqueIdPersonType(targetProperty.PropertyName, out personType))
                     {
-                        int usi = _personUniqueIdToUsiCache.GetUsi(personType, (string) queryStringParameter.Value);
+                        int usi = _personUniqueIdToUsiCache.GetUsi(personType, Convert.ToString(queryStringParameter.Value));
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:800]: why the change from the cast?

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs [hunk 3]
```diff
@@ -815,7 +815,7 @@ private void ProcessQueryStringParameters(HqlBuilderContext builderContext, Comp
                     else
                     {
                         criteriaPropertyName = targetProperty.PropertyName;
-                        parameterValue = ConvertParameterValueForProperty(targetProperty, queryStringParameter.Value);
+                        parameterValue = ConvertParameterValueForProperty(targetProperty, Convert.ToString(queryStringParameter.Value));
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:818]: should this be a cast instead?

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs (hunk 1 above)
Comment (semalaiappan) [reply Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:791]: (string) queryStringParameter.Value' threw an exception of type 'System.InvalidCastException'
    Data: {System.Collections.ListDictionaryInternal}
    HResult: -2147467262
    HelpLink: null
    InnerException: null
    Message: "Specified cast is not valid."
    Source: "3758471d3c4b41068e3d4cce19464a90"
    StackTrace: "   at <>x.<>m0(HqlBuilder <>4__this, HqlBuilderContext builderContext, CompositeDefinitionProcessorContext processorContext)"
    TargetSite: {System.String <>m0(EdFi.Ods.Features.Composites.Infrastructure.HqlBuilder, EdFi.Ods.Features.Composites.Infrastructure.HqlBuilderContext, EdFi.Ods.Features.Composites.CompositeDefinitionProcessorContext)}

Comment (semalaiappan) [reply Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:791]: So  I changed to  Convert.ToString(queryStringParameter.Value) to fix this issue

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs (hunk 2 above)
Comment (semalaiappan) [reply Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:800]: (string) queryStringParameter.Value' threw an exception of type 'System.InvalidCastException'
Data: {System.Collections.ListDictionaryInternal}
HResult: -2147467262
HelpLink: null
InnerException: null
Message: "Specified cast is not valid."
Source: "3758471d3c4b41068e3d4cce19464a90"
StackTrace: " at <>x.<>m0(HqlBuilder <>4__this, HqlBuilderContext builderContext, CompositeDefinitionProcessorContext processorContext)"
TargetSite: {System.String <>m0(EdFi.Ods.Features.Composites.Infrastructure.HqlBuilder, EdFi.Ods.Features.Composites.Infrastructure.HqlBuilderContext, EdFi.Ods.Features.Composites.CompositeDefinitionProcessorContext)}

Comment (semalaiappan) [reply Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:800]: So I changed to Convert.ToString(queryStringParameter.Value) to fix this issue

File: Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs (hunk 3 above)
Comment (semalaiappan) [reply Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:818]: (string) queryStringParameter.Value' threw an exception of type 'System.InvalidCastException'
Data: {System.Collections.ListDictionaryInternal}
HResult: -2147467262
HelpLink: null
InnerException: null
Message: "Specified cast is not valid."
Source: "3758471d3c4b41068e3d4cce19464a90"
StackTrace: " at <>x.<>m0(HqlBuilder <>4__this, HqlBuilderContext builderContext, CompositeDefinitionProcessorContext processorContext)"
TargetSite: {System.String <>m0(EdFi.Ods.Features.Composites.Infrastructure.HqlBuilder, EdFi.Ods.Features.Composites.Infrastructure.HqlBuilderContext, EdFi.Ods.Features.Composites.CompositeDefinitionProcessorContext)}

Comment (semalaiappan) [reply Application/EdFi.Ods.Features/Composites/Infrastructure/HqlBuilder.cs:818]: So I changed to Convert.ToString(queryStringParameter.Value) to fix this issue

**Review by tbmeador-dlp — APPROVED** [review]
